### PR TITLE
Add workflow to update solution files with SlnGen and update them initially

### LINKED ISSUE #42121: Issues while building trying to use the Test Explorer in certain solutions
This is all after a clean build via `./build.cmd clr+libs`. Currently on VS2019 16.8 P2.1. I've experienced this with `./build.cmd -vs System.Diagnostics.EventLog` and `./build.cmd -vs System.Globalization` but I remember seeing this with a few other solutions as well.

VS tries to build `TestUtilities` when it's asked to run tests. It looks like `TestUtilities` targets NetCoreAppCurrent and .NET Framework 4.6.1 at the same time, and for some reason it tries to build the .NET Framework version too (and fails)? This is the error I get:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0006	Metadata file 'C:\Users\gotos\source\repos\runtime\artifacts\bin\System.Runtime.CompilerServices.Unsafe\net45-Debug\System.Runtime.CompilerServices.Unsafe.dll' could not be found	TestUtilities (net5.0)	C:\Users\gotos\source\repos\runtime\src\libraries\Common\tests\TestUtilities\CSC	1	Active
```

Perhaps it should exclude the net461 unless `-framework net48` is specified when launching vs via command line? Not exactly sure what the issue is. I can provide the binlog if you need it.

cc @ViktorHofer

### LINKED ISSUE #43109: Solution Files in libraries don't list all required dependencies
Our solution files aren't up-to-date and need dependencies added to them (ie System.Runtime.CompilerServices.Unsafe). Currently most of the builds fail because of the missing Unsafe dependency which TestUtilities relies on.

The plan is to use slngen to keep the solution files up-to-date and ideally protect this in CI. To fix the build issues I will send a PR to do a manual update.

### PR DESCRIPTION
**Pull request by ViktorHofer**
Fixes https://github.com/dotnet/runtime/issues/43109
Fixes https://github.com/dotnet/runtime/issues/42121

SlnGen is used to update the solution files. Adding:
- Global tool manifest entry to slngen
- Script to update the solution files

Updating all the solution files checked in. The size increases are expected as a) SlnGen adds dedicated nodes for the solution folders and b) missing dependencies are now part of the solution file.

### CODE REVIEW

**Comment by jkotas** [comment]
Is it intentional that the .sln files are added as binary files?

**Comment by ViktorHofer** [comment]
> Is it intentional that the .sln files are added as binary files?

It's not. I'm investigating. I believe the solution files aren't generated correctly.

**Comment by ViktorHofer** [comment]
Hmm no idea:

C:\git\runtime3\src\libraries\System.Text.RegularExpressions>git check-attr -a System.Text.RegularExpressions.sln
System.Text.RegularExpressions.sln: text: set
System.Text.RegularExpressions.sln: eol: crlf

**Comment by ViktorHofer** [comment]
OK reason is, the file encoding is UTF-16LE which GH detects as binary. Will convert them and open an issue in slngen.

**Comment by ViktorHofer** [comment]
Fixed, removing the no-merge label.

**Review by Anipik — APPROVED** [review]
LGTM

File: src/libraries/UpdateLibrariesSlns.ps1 [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+'@
+
+foreach ($file in Get-ChildItem *\*.sln)
+{
+    $ProjFilePath = "$(Join-Path $file.DirectoryName $file.BaseName).proj"
+
+    $ProjContent | Out-File -FilePath $ProjFilePath
+    dotnet slngen "$ProjFilePath -p SlnGenMainProject=$($file.BaseName) --launch false --nologo"
+    Remove-Item $ProjFilePath
+}
```
Comment (akoeplinger) [inline src/libraries/UpdateLibrariesSlns.ps1:24]: could you rewrite this in msbuild so we can run it on macOS/Linux too?

Comment (ViktorHofer) [reply src/libraries/UpdateLibrariesSlns.ps1:24]: Powershell runs on macOS & Linux. We have couple other ps1 scripts which we only trigger on demand like this one.

Comment (akoeplinger) [reply src/libraries/UpdateLibrariesSlns.ps1:24]: Ah right, I keep forgetting about cross-platform powershell :D

**Pull request merged**